lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -1344,6 +1370,268 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
     return dm;
   }
 
+  static bool needScaler(ttir::GenericOp op) {
+    Block &block = op.getRegion().front();
+    Operation &firstOp = block.getOperations().front();
```

Comment:
Technically we probably need to walk the block to search for a reduce, but this is OK for now.

Reply:
Agreed, this was a temporary solution anyway.